packages/wrangler/src/init.ts

```diff
@@ -485,9 +486,17 @@ export async function initHandler(args: InitArgs) {
 					`/accounts/${accountId}/workers/services/${fromDashScriptName}/environments/${defaultEnvironment}/content`
 				);
 
-				await writeFile(
-					path.join(creationDirectory, "./src/index.ts"),
-					dashScript
+				// TODO: writeFile in small batches to not exhaust system file descriptors
+				await Promise.all(
+					dashScript.map(async (file) => {
+						const filepath = path
+							.join(creationDirectory, `./src/${file.name}`)
+							.replace(/\.js$/, ".ts"); // change javascript extension to typescript extension
+						const directory = dirname(filepath);
+
+						await mkdir(directory, { recursive: true });
+						await writeFile(filepath, file.stream() as ReadableStream);
+					})
```

Comment:
So something like this maybe? 
```suggestion
function chunk(array, size) {
  const result = [];
  let i = 0;

  while (i < array.length) {
    result.push(array.slice(i, i + size));
    i += size;
  }

  return result;
}

const batchSize = 10;
const batches = chunk(dashScript, batchSize);

await Promise.all(
  batches.map(async (batch) => {
    await Promise.all(
      batch.map(async (file) => {
        const filepath = path
          .join(creationDirectory, `./src/${file.name}`)
          .replace(/\.js$/, ".ts"); // change javascript extension to typescript extension
        const directory = dirname(filepath);

        await mkdir(directory, { recursive: true });
        await writeFile(filepath, file.stream() as ReadableStream);
      })
    );
  })
);
```